File: day:78/zigZag.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
    // base cases
    if(s === '') return ''
    if(numRows === 1) return s;
    const STEP = (2 * numRows) - 2;
    const result = [];
    for(let i = 0; i < numRows; i++) {
        let j = i;
        while(j < s.length) {
            result.push(s[j]);
            // middle characters
            if(i !== 0 && i !== (numRows - 1) && (j + STEP - 2 * i) < s.length) {
                result.push(s[j + STEP - 2 * i])
            }
            j += STEP;
        }
    }
    return result.join('')
};

File: day:78/zigZag.rs
let mut step = if num_rows == 1 { 0 } else { - 1 };
	let mut rows: Vec<String> = vec![String::new(); num_rows as usize];
	let mut i = 0i32;

	for c in s.chars() {
		rows[i as usize].push(c);
		if i == 0 || i == (num_rows as i32 - 1) {
			step = -step;
		}
		i += step;
	}

	rows.join("")
